batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,137 @@
+        }
+
+    }
+
+    @Override
+    public void add(User user) throws DataException {
+        String str = "insert into user (firstname,surname,username,password) values (?,?,?,?)";
+
+        try (PreparedStatement statement = connection.prepareStatement(str);) {
+
+            statement.setString(1, user.getFirstName());
+            statement.setString(2, user.getSurName());
+            statement.setString(3, user.getUserName());
+            statement.setString(4, user.getPassword());
+
+            statement.execute();
+        } catch (SQLException e) {
+            logger.log(Level.SEVERE, e.getMessage());
+            throw new DataException();
+        }
+    }
+
+    @Override
+    public void delete(int userID) throws DataException {
+        String query = "delete from user where id= ?";
+
+        try (PreparedStatement statement = connection.prepareStatement(query);) {
+            statement.setLong(1, userID);
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            logger.log(Level.SEVERE, e.getMessage());
+            throw new DataException(e.getMessage());
+        }
+    }
+
+    @Override
+    public List<User> fetch(int page) throws DataException {
+        String query = "select * from user";
+        List<User> userList = new ArrayList<User>();
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet results = statement.executeQuery();
+
+            while (results.next()) {
+                User user = new User();
+
+                user.setId(TypeCaster.toInt(results.getString("id")));
+                user.setFirstName(results.getString("firstname"));
+                user.setSurName(results.getString("surname"));
+                user.setUserName(results.getString("username"));
+                user.setPassword(results.getString("password"));
+
+                userList.add(user);
+
+            }
+
+        } catch (SQLException e) {
+            logger.log(Level.SEVERE, e.getMessage());
+            throw new DataException(e.getMessage());
+        }
+        return userList;
+
+    }
+
+    @Override
+    public User fetchByID(int userID) throws DataException {
+        User user = null;
+        String query = "select * from user where id=?";
+
+        try (PreparedStatement statement = connection.prepareStatement(query);) {
+            statement.setLong(1, userID);
+            ResultSet results = statement.executeQuery();
+
+            while (results.next()) {
+                user = new User();
+
+                user.setId(TypeCaster.toInt(results.getString("id")));
+                user.setFirstName(results.getString("firstname"));
+                user.setSurName(results.getString("surname"));
+                user.setUserName(results.getString("username"));
+                user.setPassword(results.getString("password"));
+            }
+
+        } catch (SQLException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)